components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -710,16 +711,19 @@ export class TypeORMWorkspaceDBImpl extends TransactionalDBImpl<WorkspaceDB> imp
 
     // Find the (last triggered) prebuild for a given commit
     public async findPrebuiltWorkspaceByCommit(
-        cloneURL: string,
+        projectId: string,
         commit: string,
     ): Promise<PrebuiltWorkspace | undefined> {
-        if (!commit || !cloneURL) {
-            return undefined;
+        if (!commit || !projectId) {
+            throw new ApplicationError(ErrorCodes.INTERNAL_SERVER_ERROR, "Illegal arguments", { projectId, commit });
```

Comment:
we should not silently ignore such a state

Reply:
Agree, but how comes that this is necessary in the persistence layer given the mandatory arguments. Let's check the call sites, too.